EIPS/eip-2535.md

```diff
@@ -0,0 +1,465 @@
+    /// @dev If facet is not found return address(0).
+    /// @param _functionSelector The function selector.
+    /// @return The facet address.
+    function facetAddress(bytes4 _functionSelector) external view returns(address);
+}
+```
+See the [reference implementation](https://github.com/mudgen/Diamond) to see an example of how this can be implemented.
+
+These functions are designed for user-interface software. A user interface calls these functions to provide information about and visualize diamonds.
+
+The `facetFunctionsSelectors`, `facets`, `facetAddresses` functions are designed to be called by off-chain software and so are not gas efficient.
+
+### Design Points
+
+A diamond implements the following design points:
+
+1. A diamond contains a fallback function, a constructor, and zero or more immutable functions that are defined directly within it.
+2. The `diamondCut` function is used to add/replace/remove functions in the diamond. The `diamondCut` function can be an "immutable function" that is defined directly in the diamond or it can be defined in a facet. 
+3. The `diamondCut` function associates function selectors with facets and emits the `DiamondCut` event.
+4. When a function is called on a diamond it executes immediately if it is an "immutable function" defined directly in the diamond. Otherwise the diamond's fallback function is executed. The fallback function finds the facet associated with the function and executes the function using `delegatecall`. If there is no facet for the function then execution reverts.
+5. The `bytes[] memory _diamondCut` argument to the `diamondCut` function specifies the functions to add, replace and remove. If the facet address to associate function selectors with is `address(0)` then the provided functions selectors are removed from the diamond, otherwise the function selectors are added or replaced in the diamond. The `diamondCut` function reverts if a supplied function selector is already associated with the supplied facet address. The `diamondCut` function reverts if a supplied function selector can't be removed because it already does not exist.
+6. Each time functions are added, replaced or removed a `DiamondCut` event is emitted to record it.
+7. A diamond implements ERC165. If a diamond has the `diamondCut` function then the interface ID used for it is Diamond.diamondCut.selector. The interface ID used for the DiamondLoupe interface is `DiamondLoupe.facets.selector ^ DiamondLoupe.facetFunctionSelectors.selector ^ DiamondLoupe.facetAddresses.selector ^ DiamondLoupe.facetAddress.selector`.
+
+This standard does not prevent other functions from being added that add, replace or remove functions. But in any case the `DiamondCut` event must be emitted for any addition, replacement or removal of functions.
+
+The diamond address is the address that users interact with. The diamond address does not change. Only facet addresses can change by using the `diamondCut` function.
+
+## Diamond Example
```

Comment:
This doesn't follow the EIP template section headings.  Recommend putting this as a sub-heading in one of the template sections.

Reply:
Perhaps move this to the implementations section, or link to it from the implementations section?

Reply:
Yes. I moved the example under the Implementation section.